src/medVtkInria/vtkDataManagement/vtkMetaDataSet.cxx

```diff
@@ -655,70 +574,23 @@ void vtkMetaDataSet::CopyInformation (vtkMetaDataSet* metadataset)
 }
 
 //----------------------------------------------------------------------------
-double vtkMetaDataSet::GetScalarNullValue(const char * arrayName)
-{
-    if ( ! this->GetDataSet())
-    return NAN;
-
-    vtkDataArray * array = this->GetDataSet()->GetPointData()->GetArray(arrayName);
-    if ( ! array)
-        array = this->GetDataSet()->GetCellData()->GetArray(arrayName);
-    if ( ! array || ScalarNullValues.find(arrayName) == ScalarNullValues.end())
-        return NAN;
-
-    return ScalarNullValues[arrayName];
-}
-
-
-//----------------------------------------------------------------------------
-void vtkMetaDataSet::SetScalarNullValue(const char * arrayName, double nullValue)
-{
- if ( ! this->GetDataSet())
-    return;
-
- vtkDataArray * array = this->GetDataSet()->GetPointData()->GetArray(arrayName);
- if ( ! array)
-    array = this->GetDataSet()->GetCellData()->GetArray(arrayName);
- if ( ! array)
-    return;
-
- ScalarNullValues[arrayName] = nullValue;
-}
 
-//----------------------------------------------------------------------------
-double* vtkMetaDataSet::GetScalarRange(QString attributeName)
+double* vtkMetaDataSet::GetCurrentScalarRange()
```

Comment:
Why did you remove the modifications done in https://github.com/Inria-Asclepios/medInria-public/pull/338 ?

Reply:
it has been complicated to know what changes we have done and want to keep/or not, or done by medInria and want to keep/or not :D. I was looking at our PRs this morning, and tested this part. I added back some of your changes, but i'm going to check the exact PR, thank you

Reply:
Good work and good luck with all the changes  :-)

Reply:
I just checked if the PR was still useful and that is the case we need it for 4D meshes

Reply:
I tested with a 4D mesh and do not find any problem. What do i need to do to see the bug?

Reply:
You need a 4D mesh with more than one attribute (I will send you one). When you change attributes from the default attribute  and you click  on modify range you want the min and max value of the range to correspond to the actual attributes

Reply:
Thanks for the 4D mesh with multiple attributes, the PR is updated :)